FIX: Postgres create and delete materialized view

krystxf opened the pull request:
this fixes some issues from #1461 

- fix `SQL: create` in context menu functionality for materialized views
- fix `Drop` in context menu functionality for materialized views

rathboma commented on apps/studio/src/lib/db/client.ts:
```diff
@@ -484,6 +487,11 @@ function getViewCreateScript(server: IDbConnectionServer, database: IDbConnectio
   return database.connection?.getViewCreateScript(view);
 }
 
+function getMaterializedViewCreateScript(server: IDbConnectionServer, database: IDbConnectionDatabase, view: string /* , schema */) {
```
Given that this is only implemented for postgresql, this function needs to check if this function exists before calling it, and otherwise return a default value, probably `null`.

rathboma commented on apps/studio/src/lib/db/clients/postgresql.ts:
```diff
@@ -1372,7 +1385,7 @@ export async function truncateAllTables(conn: Conn, schema: string) {
   await runWithConnection(conn, async (connection) => {
     const connClient = { connection };
-    const sql = `DROP ${PD.wrapLiteral(typeOfElement)} ${wrapIdentifier(schema)}.${wrapIdentifier(elementName)}`
+    const sql = `DROP ${PD.wrapLiteral(DatabaseElement[typeOfElement])} ${wrapIdentifier(schema)}.${wrapIdentifier(elementName)}`
```
Forgive my ignorance here - how is this change different in function to the prior version?

krystxf replied:
it works like a key-value this way, it didn't make any difference since `typeOfElement` were same as values (one word), but with materialized view `typeOfElement` is `materialized-view`, but value that has to be filled in here is `MATERIALIZED VIEW`, I thought this was the easiest way how fix it.

rathboma replied:
Ok, sounds good. Looking forward to the other minor changes then will merge :+1:

rathboma commented on apps/studio/src/lib/db/client.ts:
```diff
@@ -64,6 +65,7 @@ export interface DatabaseClient {
   getTableProperties: (table: string, schema?: string) => Promise<TableProperties | null>,
   getTableCreateScript: (table: string, schema?: string) => Promise<string>,
   getViewCreateScript: (view: string) => void,
+  getMaterializedViewCreateScript: (view: string) => void,
```
This should return the type `Promise<String>`, and it should be optional also:

`getMaterializedViewCreateScript?:`, similar to `defaultSchema`

krystxf replied:
The function is pretty similar to `getViewCreateScript` function so I used the same types, isn't  in that case mistake also in `getViewCreateScript` function?

rathboma replied:
Yeah the interface is wrong for many functions! A contributor converted us to TS and some stuff is half converted :-)

krystxf replied:
so is this conversation resolved or should I change something?

rathboma reviewed (changes requested):
Thank you so much for your PR :pray: .

I have a couple of minor changes I'd like for you to make before I merge it, but this is an awesome contribution. Thanks so much.

krystxf commented:
Here [e5e55ca](https://github.com/beekeeper-studio/beekeeper-studio/pull/1502/commits/e5e55cad649ce2e545a7b937e44769a734d845e5) I added check if function exists

I also added a function to the other databases files (`mysql.ts`, etc.), following the pattern of how functions that the database does not have were handled

rathboma reviewed (approved):
You rock! Thanks so much for this
